pkg/cloudprovider/types.go

```diff
@@ -88,10 +89,18 @@ type InstanceType struct {
 	// Overhead is the amount of resource overhead expected to be used by kubelet and any other system daemons outside
 	// of Kubernetes.
 	Overhead *InstanceTypeOverhead
+
+	once        sync.Once
+	allocatable v1.ResourceList
+}
+
+func (i *InstanceType) precompute() {
+	i.allocatable = resources.Subtract(i.Capacity, i.Overhead.Total())
 }
 
 func (i *InstanceType) Allocatable() v1.ResourceList {
-	return resources.Subtract(i.Capacity, i.Overhead.Total())
+	i.once.Do(i.precompute)
+	return i.allocatable.DeepCopy()
```

Comment:
Do we need this `DeepCopy` or can we assume that we shouldn't be modifying this readonly value?

Reply:
I just err'd on the side of safety here, it's still faster than before.